bliss/surveys/sdss_download.py

```diff
@@ -4,70 +4,98 @@
-    with open(filename, "wb") as out_file:
-        out_file.write(bz2.decompress(response.content))
+def download_frame(run=94, camcol=1, field=12, band="r", download_dir="."):
+    download_file(
+        f"{URLBASE}/photo/redux/301/{_stripped(run)}/objcs/{camcol}/"
+        f"fpM-{_run6(run)}-{band}{camcol}-{_field4(field)}.fit.gz",
+        f"{download_dir}/{_subdir3(run, camcol, field)}/"
+        f"fpM-{_run6(run)}-{band}{camcol}-{_field4(field)}.fits",
+        gzip.decompress,
+    )
+
+    download_file(
+        f"{URLBASE}/photoObj/frames/301/{_stripped(run)}/{camcol}/"
+        f"frame-{band}-{_run6(run)}-{camcol}-{_field4(field)}.fits.bz2",
+        f"{download_dir}/{_subdir3(run, camcol, field)}/"
+        f"frame-{band}-{_run6(run)}-{camcol}-{_field4(field)}.fits",
+        bz2.decompress,
+    )
+
+
+def download_psfield(run=94, camcol=1, field=12, download_dir="."):
+    download_file(
+        f"{URLBASE}/photo/redux/301/{_stripped(run)}/objcs/{camcol}/"
+        f"psField-{_run6(run)}-{camcol}-{_field4(field)}.fit",
+        f"{download_dir}/{_subdir3(run, camcol, field)}/"
+        f"psField-{_run6(run)}-{camcol}-{_field4(field)}.fits",
+    )
 
 
 def download_all(run=94, camcol=1, field=12, download_dir="."):
     if not Path(download_dir).exists():
         # create download directory
         Path(download_dir).mkdir(parents=True, exist_ok=True)
 
-    # strip leading zeros
-    run_stripped = str(run).lstrip("0")
-    field_stripped = str(field).lstrip("0")
+    download_pf(run, camcol, download_dir)
+    download_po(run, camcol, field, download_dir)
 
-    run6 = f"{int(run_stripped):06d}"
-    field4 = f"{int(field_stripped):04d}"
+    for band in BANDS:
+        download_frame(run, camcol, field, band, download_dir)
 
-    subdir2 = f"{run_stripped}/{camcol}"
-    subdir3 = f"{run_stripped}/{camcol}/{field_stripped}"
+    download_psfield(run, camcol, field, download_dir)
 
-    download_file(
-        f"{URLBASE}/photoObj/301/{run_stripped}/photoField-{run6}-{camcol}.fits",
-        f"{download_dir}/{subdir2}/photoField-{run6}-{camcol}.fits",
-    )
-    download_file(
-        f"{URLBASE}/photoObj/301/{run_stripped}/{camcol}/photoObj-{run6}-{camcol}-{field4}.fits",
-        f"{download_dir}/{subdir3}/photoObj-{run6}-{camcol}-{field4}.fits",
-    )
 
-    for band in BANDS:
-        download_gz_file(
-            f"{URLBASE}/photo/redux/301/{run_stripped}/objcs/{camcol}/"
-            f"fpM-{run6}-{band}{camcol}-{field4}.fit.gz",
-            f"{download_dir}/{subdir3}/fpM-{run6}-{band}{camcol}-{field4}.fits",
-        )
-
-        download_bz2_file(
-            f"{URLBASE}/photoObj/frames/301/{run_stripped}/{camcol}/"
-            f"frame-{band}-{run6}-{camcol}-{field4}.fits.bz2",
-            f"{download_dir}/{subdir3}/frame-{band}-{run6}-{camcol}-{field4}.fits",
-        )
+# Helper functions
+def download_file(url, filename, preprocess_fn=lambda x: x):  # noqa: WPS404
```

Comment:
Please remember to delete `data/sdss/Makefile` once we're satisified with this download routine